test/battle/move_effect/bug_bite.c

```diff
@@ -0,0 +1,137 @@
+#include "global.h"
+#include "test/battle.h"
+
+ASSUMPTIONS
+{
+    ASSUME(gBattleMoves[MOVE_BUG_BITE].effect == EFFECT_BUG_BITE);
+    ASSUME(gBattleMoves[MOVE_BUG_BITE].pp == 20);
+}
+
+// Pretty much copy/paste of the Berry Fling Test.
+SINGLE_BATTLE_TEST("Bug Bite eats the target's berry and immediately gains its effect")
+{
+    u16 item;
+    u32 status1 = STATUS1_NONE, effect, statId;
+
+    PARAMETRIZE {item = ITEM_NONE; }
+    PARAMETRIZE { item = ITEM_ORAN_BERRY; effect = HOLD_EFFECT_RESTORE_HP; }
+    PARAMETRIZE { item = ITEM_SITRUS_BERRY; effect = HOLD_EFFECT_RESTORE_HP; }
+    // PARAMETRIZE { item = ITEM_ENIGMA_BERRY; effect = HOLD_EFFECT_RESTORE_HP; } To do once Enigma Berry's effect is done
+    PARAMETRIZE { item = ITEM_LEPPA_BERRY; effect = HOLD_EFFECT_RESTORE_PP; }
+    PARAMETRIZE { item = ITEM_CHESTO_BERRY; effect = HOLD_EFFECT_CURE_SLP; status1 = STATUS1_SLEEP; }
+    PARAMETRIZE { item = ITEM_CHERI_BERRY; effect = HOLD_EFFECT_CURE_PAR; status1 = STATUS1_PARALYSIS; }
+    PARAMETRIZE { item = ITEM_PECHA_BERRY; effect = HOLD_EFFECT_CURE_PSN; status1 = STATUS1_POISON; }
+    PARAMETRIZE { item = ITEM_PECHA_BERRY; effect = HOLD_EFFECT_CURE_PSN; status1 = STATUS1_TOXIC_POISON; }
+    PARAMETRIZE { item = ITEM_RAWST_BERRY; effect = HOLD_EFFECT_CURE_BRN; status1 = STATUS1_BURN; }
+    // PARAMETRIZE { item = ITEM_ASPEAR_BERRY; effect = HOLD_EFFECT_CURE_FRZ; status1 = STATUS1_FREEZE; } Can't use a move while frozen.
```

Comment:
Maybe we need a comment for Frostbite, since there's no move with Frostbite by default.

Reply:
Actually, no. The battler can start with frostbite, so it should be added here as well.

Reply:
This still needs addressing 👀

Reply:
Done, also fixed a bug in the test runner, because it had status1 stored only as u8 and STATUS1_FROSTBITE didn't fit.